samples/retained-instance-store/src/main/java/com/badoo/ribs/samples/retained_instance_store/rib/retainedInstance/RetainedInstancePresenter.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance
+
+import android.os.Parcelable
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.android.subscribe
+import com.badoo.ribs.core.plugin.ViewAware
+import com.badoo.ribs.routing.source.backstack.BackStack
+import com.badoo.ribs.routing.source.backstack.operation.replace
+import com.badoo.ribs.samples.retained_instance_store.rib.retainedInstance.routing.RetainedInstanceRouter
+import java.lang.IllegalStateException
+
+interface RetainedInstancePresenter {
+    fun toggleChildCounter()
+}
+
+internal class RetainedInstancePresenterImpl(
+        private val backStack: BackStack<RetainedInstanceRouter.Configuration>
+) : ViewAware<RetainedInstanceView>,
+        RetainedInstancePresenter {
+
+    private var view: RetainedInstanceView? = null
```

Comment:
1. It's a bad idea to store a reference to it (even if you now null it correctly)
2. It looks like you're not using the reference at all?